stable_baselines3/common/vec_env/vec_extract_dict_obs.py

```diff
@@ -13,14 +14,19 @@ class VecExtractDictObs(VecEnvWrapper):
 
     def __init__(self, venv: VecEnv, key: str):
         self.key = key
+        assert isinstance(
+            venv.observation_space, spaces.Dict
+        ), f"VecExtractDictObs can only be used with Dict obs space, not {venv.observation_space}"
         super().__init__(venv=venv, observation_space=venv.observation_space.spaces[self.key])
 
     def reset(self) -> np.ndarray:
         obs = self.venv.reset()
+        assert isinstance(obs, dict)
```

Comment:
in the future, we should probably use [`Generic`](https://docs.python.org/3/library/typing.html#typing.Generic) for `VecEnvWrapper` and `VecEnvWrapper` to handle this case.

Reply:
I agree, my goal for now is to remove as many files as possible from mypy ignore and then do a second pass later to polish the edges (i.e. better typing hints).